classes/WP_CLI/Action/List.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * WP-CLI command: action-scheduler action list
+ */
+class ActionScheduler_WPCLI_Action_List_Command extends ActionScheduler_WPCLI_Command {
+
+	const PARAMETERS = array(
+		'hook',
+		'args',
+		'date',
+		'date_compare',
+		'modified',
+		'modified_compare',
+		'group',
+		'status',
+		'claimed',
+		'per_page',
+		'offset',
+		'orderby',
+		'order',
+	);
+
+	/**
+	 * Execute command.
+	 *
+	 * @return void
+	 */
+	public function execute() {
+		$store  = \ActionScheduler::store();
+		$logger = \ActionScheduler::logger();
+
+		$fields = array(
+			'id',
+			'hook',
+			'status',
+			'group',
+			'recurring',
+			'scheduled_date',
+		);
+
+		$this->process_csv_arguments_to_arrays();
+
+		if ( ! empty( $this->assoc_args['fields'] ) ) {
+			$fields = $this->assoc_args['fields'];
+		}
+
+		$formatter = new \WP_CLI\Formatter( $this->assoc_args, $fields );
+
+		$query_args = array_filter(
+			$this->assoc_args,
+			static function ( $key ) {
+				return in_array( $key, static::PARAMETERS, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
```

Comment:
A bit shorter and possibly more efficient to do either:

```suggestion
		$query_args = wp_array_slice_assoc( $this->assoc_args, static::PARAMETERS );
```

Or:

```suggestion
		$query_args = array_intersect_key( $assoc_args, array_flip( static::PARAMETERS ) );
```

Reply:
Thinking on it some more, I wonder if we should even have this check/filter.

- Should we subsequently add support for new query args, this means we need updates in two different places.
- If a new or custom data store supports additional query args of its own, they won't be supported through this interface.